snuba/util.py

```diff
@@ -325,3 +329,21 @@ def create_metrics(prefix: str, tags: Optional[Tags] = None) -> MetricsBackend:
             else None,
         ),
     )
+
+
+def with_span(op="function", var=None):
+    def decorator(func):
+        frame_info = inspect.stack()[1]
+        filename = frame_info.filename
```

Comment:
I'd be curious what the runtime overhead of this is. (I don't have any concrete reason to believe it's _bad_, but seems like it could be.)

Reply:
I had a similar concern about leaking a stack frame inside a scope but then I realized there'd be, like, a dozen of them.